joplin/pages/event_page/models.py

```diff
@@ -25,13 +25,31 @@
 from publish_preflight.requirements import FieldPublishRequirement, StreamFieldPublishRequirement
 
 
+def only_one_physical_location(stream_value):
+    """
+    Streamfield requirement
+    :param stream_value:
+    Checks to see that the stream_value has length and if there are two locations that one of them is a virtual event
+    Events cannot have two physical locations
+    """
+    if not stream_value:
+        return False
+    if len(stream_value) > 0:
+        if len(stream_value.stream_data) == 2:
+            loc1 = stream_value.stream_data[0][0]
+            loc2 = stream_value.stream_data[1][0]
+            return loc1 == 'virtual_event' or loc2 == 'virtual_event'
```

Comment:
if neither of the locations is a virtual event, then this will return False, which would prevent the page from publishing. otherwise, it's fine.

Reply:
👍